benchmarks/src/main.rs

```diff
@@ -15,106 +14,148 @@ macro_rules! spawn_blocking_and_await {
-    let score = task_size / (median.as_millis() as f64 / 1000.0);
-    println!(
-        "TEST: {} -- Score: {}, Median Iteration Time: {:.3}ms\n",
-        name,
-        score,
-        median.as_secs_f64()
-    );
-
-    if more_info {
-        println!(
-            "10th percentile: {:#?}",
-            get_nth_percentile(&durations, 10.0),
-        );
-        println!(
-            "25th percentile: {:#?}",
-            get_nth_percentile(&durations, 25.0),
-        );
-        println!(
-            "50th percentile: {:#?}",
-            get_nth_percentile(&durations, 50.0),
-        );
-        println!(
-            "75th percentile: {:#?}",
-            get_nth_percentile(&durations, 75.0),
-        );
-        println!(
-            "90th percentile: {:#?}",
-            get_nth_percentile(&durations, 90.0),
-        );
-        println!(
-            "95th percentile: {:#?}",
-            get_nth_percentile(&durations, 95.0),
-        );
-        println!(
-            "98th percentile: {:#?}",
-            get_nth_percentile(&durations, 98.0),
-        );
-        println!(
-            "99th percentile: {:#?}\n",
-            get_nth_percentile(&durations, 99.0),
-        );
-    }
-
-    score
-}
-
-async fn single_doc_benchmarks(uri: &str, more_info: bool, ids: &[bool]) -> Result<f64> {
-    println!("----------------------------");
-    println!("Single-Doc Benchmarks:");
-    println!("----------------------------\n");
-
-    let mut comp_score: f64 = 0.0;
-    let mut benchmark_count = 0_usize;
+// benchmark names
+const FLAT_BSON_ENCODING: &'static str = "Flat BSON Encoding";
+const FLAT_BSON_DECODING: &'static str = "Flat BSON Decoding";
+const DEEP_BSON_ENCODING: &'static str = "Deep BSON Encoding";
+const DEEP_BSON_DECODING: &'static str = "Deep BSON Decoding";
+const FULL_BSON_ENCODING: &'static str = "Full BSON Encoding";
+const FULL_BSON_DECODING: &'static str = "Full BSON Decoding";
+const RUN_COMMAND_BENCH: &'static str = "Run Command";
+const FIND_ONE_BENCH: &'static str = "Find one";
+const FIND_MANY_BENCH: &'static str = "Find many and empty cursor";
+const GRIDFS_DOWNLOAD_BENCH: &'static str = "GridFS download";
+const LDJSON_MULTI_EXPORT_BENCH: &'static str = "LDJSON multi-file export";
+const GRIDFS_MULTI_DOWNLOAD_BENCH: &'static str = "GridFS multi-file download";
+const SMALL_DOC_INSERT_ONE_BENCH: &'static str = "Small doc insertOne";
+const LARGE_DOC_INSERT_ONE_BENCH: &'static str = "Large doc insertOne";
+const SMALL_DOC_BULK_INSERT_BENCH: &'static str = "Small doc bulk insert";
+const LARGE_DOC_BULK_INSERT_BENCH: &'static str = "Large doc bulk insert";
+const GRIDFS_UPLOAD_BENCH: &'static str = "GridFS upload";
+const LDJSON_MULTI_IMPORT_BENCH: &'static str = "LDJSON multi-file import";
+const GRIDFS_MULTI_UPLOAD_BENCH: &'static str = "GridFS multi-file upload";
+
+/// Benchmarks included in the "BSONBench" composite.
+const BSON_BENCHES: &[&'static str] = &[
```

Comment:
the breakdown of composite scores that we were using wasn't fully accurate, so I sorted that out here with these constants.